SparkPost/2web2ui
# SA-2345 - Adds custom useStateMachine hook (Subtask 😬 for SA-2216)

[SA-2345](https://sparkpost.atlassian.net/browse/SA-2345) - a subtask of [SA-2216](https://sparkpost.atlassian.net/browse/SA-2216)

### What Changed

- Adds new custom `useStateMachine` hook with supporting tests and documentation

### How To Test

- Verify tests adequately test new hook
- Verify docs are understandable

### To Do
